app/src/main/java/me/saket/dank/utils/Intents.java

```diff
@@ -81,6 +82,16 @@ public static Intent createForPlayStoreListing(Context context, String packageNa
     }
   }
 
+  @Nullable
+  public static Intent createForYouTube(String videoUrl) {
+      String videoId = UrlParser.getVideoIDFromYouTubeUrl(videoUrl);
+      if (videoId != null) {
+          String formattedYouTuneIntentURL = "vnd.youtube://" + videoId;
+          return Intents.createForOpeningUrl(formattedYouTuneIntentURL);
+      } else
+          return null;
+  }
```

Comment:
Returning null from this function doesn't look like a good pattern. If it's called `createForYouTube()` then it should be the caller's responsibility to ensure this only gets called for YouTube URLs. I have some ideas for this code. Continuing this discussion at the call-site of this function.